civit_image_downloader.py

```
@@ -578,12 +595,19 @@ async def _write_meta_data(self, meta: Optional[Dict[str, Any]], base_output_pat
              self.logger.error(f"Error writing metadata to {output_path_final}: {e}", exc_info=True); return False, None
 
     # --- API Fetching Method ---
-    @retry( 
-        stop=stop_after_attempt(1 + parse_arguments().retries),
-        wait=wait_random_exponential(multiplier=1, max=10),
-        retry=retry_if_exception(should_retry_exception),
-        before_sleep=before_sleep_log(retry_logger, logging.WARNING)
+
+    @retry(
+    # Only retry if the error is specifically a 429
+    retry=retry_if_exception(lambda e: isinstance(e, requests.HTTPError) and e.response.status_code == 429),
```

Ruff (0.14.10)

Undefined name `requests`

(F821)